libcontainer/cgroups/fs2/memory.go

```diff
@@ -57,7 +57,10 @@ func setMemory(dirPath string, r *configs.Resources) error {
 	// never write empty string to `memory.swap.max`, it means set to 0.
 	if swapStr != "" {
 		if err := cgroups.WriteFile(dirPath, "memory.swap.max", swapStr); err != nil {
-			return err
+			// If swap is not enabled, silently ignore setting to max or disabling it.
+			if !(errors.Is(err, os.ErrNotExist) && (swapStr == "max" || swapStr == "0")) {
```

Comment:
Your PR has indeed fixed the issue, but I think we should change the behavior according to what's doing in cgroup v1: https://github.com/opencontainers/runc/blob/59ca3a818c41087ecab64286caac80df9257161e/libcontainer/cgroups/fs/memory.go#L72-L77

Reply:
Thanks! This is essentially what my code is doing, except
 - no extra syscalls, so a tad faster;
 - not dependent on memory setting (in fact it is, but this is done earlier)
 
 What aspect of it do you think needs to be changed?

Reply:
> What aspect of it do you think needs to be changed?

In cgroup v1, we check whether the `memory.memsw.limit_in_bytes` exists or not; but in cgroup v2, we always write `memory.swap.max` once the `swapStr` is no empty. I think we should change it to check the exist of `memory.swap.max`, if it is not exist, set the `swapStr` to empty, instead of check the write error type of `memory.swap.max`.

Reply:
This was done that way in cgroup v1 because cgroup management for swap was added later, and for quite some time the default expectation was "we don't have cgroup swap management".

For cgroup v2, the default expectation is "we have swap management", so we assume that `memory.swap.max` is available. If not, no big deal.

The (very slight) difference between the two implementations is what do we expect by default. The end result is the same in both cases.

The way this is implemented in cgroup v1 should not be seen as the "golden standard". Systems are evolving, defaults are changing, etc. More to say, at some point we're going to remove cgroup v1 code entirely (I very much hope to see that day).

Reply:
Ok